packages/blocks/src/api/utils.js

```diff
@@ -38,7 +38,12 @@ export function isUnmodifiedDefaultBlock( block ) {
 		return false;
 	}
 
-	const newDefaultBlock = createBlock( defaultBlockName );
+	// Cache a newly created default block.
+	if ( ! isUnmodifiedDefaultBlock.block ) {
+		isUnmodifiedDefaultBlock.block = createBlock( defaultBlockName );
+	}
```

Comment:
While this is a good improvement for the moment, at some point, I'd like us to move more towards reactive patterns. For instance if the we set a new defaultBlockName, this should be invalidated... (but this will be done in selectors at that time I guess).

Reply:
Oh, forgot about that. We can also take it into consideration here.